Objective-C/Internal/Foundation+CBL.h

```diff
@@ -0,0 +1,36 @@
+//
+//  Foundation+CBL.h
+//  CouchbaseLite
+//
+//  Copyright (c) 2020 Couchbase, Inc All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+#import "c4Base.h"
+#import "fleece/slice.hh"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface NSString (CBL)
+- (C4Slice) c4slice;
```

Comment:
We have been using CBLStringBytes for this. Can we use CBLStringBytes instead?

Reply:
CBLStringBytes usage needs a declaration and usage.  Normally two lines, one variable declaration and then use it in c4 APIs. 
```
CBLStringBytes varSlice(platformString)
c4db_setCookies(..., varSlice, ...)
```

If we use this, its more easier to use
```
c4db_setCookies(..., platformString.c4Slice, ...)
```

Reply:
CBLStringBytes has also some optimization. I don't think we should have two ways of converting the NString <-> Slice if possible. Otherwise we will have a question to ourselves which one to use and why.

Reply:
Ok. let me check, if i can consume CBLStringBytes, so we can achieve the optimization. I will anyways remove from this PR and create a separate PR